zipkin-storage/zipkin2_cassandra/src/main/resources/zipkin2-schema.cql

```diff
@@ -0,0 +1,108 @@
+CREATE KEYSPACE IF NOT EXISTS zipkin2
+  WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}
+  AND durable_writes = false;
+
+CREATE TYPE IF NOT EXISTS zipkin2.endpoint (
+    service      text,
+    ipv4         inet,
+    ipv6         inet,
+    port         int,
+);
+
+CREATE TYPE IF NOT EXISTS zipkin2.annotation (
+    ts bigint,
+    v  text,
+);
+
+CREATE TABLE IF NOT EXISTS zipkin2.span (
+    trace_id            text, // when strictTraceId=false, only contains right-most 16 chars
+    ts_uuid             timeuuid,
+    id                  text,
+    trace_id_high       text, // when strictTraceId=false, contains left-most 16 chars if present
+    parent_id           text,
+    kind                text,
+    span                text, // span.name
```

Comment:
@michaelsembwever @llinder I noticed we have no SASI on this, which afaict means that we can't do a search for span name when service isn't present.. what should we do about this?

Reply:
if you need to search by span name without service name, you'd do this via the `trace_by_service_span` table.
you'd do this via writing `(("", span_name, bucket), timestamp_uuid)` rows…

but… take care of hot partitions: if there's not a high enough cardinality on span_name then there's going to be hot partitions on those `("",span_name)` combinations. hot partitions affect write throughput and latency. it can be addressed by putting in a write cache like the `DeduplicatingExecutor`.